tests/LanguageServerCodeTransform/Unit/LspCommand/GenerateMethodCommandTest.php

```diff
@@ -0,0 +1,173 @@
+<?php
+
+namespace Phpactor\Extension\LanguageServerCodeTransform\Tests\Unit\LspCommand;
+
+use Amp\Promise;
+use function Amp\Promise\wait;
+use Phpactor\CodeTransform\Domain\Exception\TransformException;
+use Phpactor\WorseReflection\Core\Exception\MethodCallNotFound;
+use Phpactor\CodeTransform\Domain\Refactor\GenerateMethod;
+use Phpactor\CodeTransform\Domain\SourceCode;
+use Phpactor\Extension\LanguageServerBridge\Converter\TextEditConverter;
+use Phpactor\Extension\LanguageServerCodeTransform\LspCommand\GenerateMethodCommand;
+use Phpactor\LanguageServer\Core\Server\ClientApi;
+use Phpactor\LanguageServer\Core\Server\RpcClient;
+use Phpactor\LanguageServer\Core\Workspace\Workspace;
+use Phpactor\LanguageServerProtocol\ApplyWorkspaceEditResponse;
+use Phpactor\LanguageServerProtocol\TextDocumentItem;
+use Phpactor\LanguageServerProtocol\WorkspaceEdit;
+use Phpactor\TextDocument\TextDocumentEdits;
+use Phpactor\TextDocument\TextDocumentUri;
+use Phpactor\TextDocument\TextEdit;
+use Phpactor\TextDocument\TextEdits;
+use PHPUnit\Framework\TestCase;
+use Prophecy\PhpUnit\ProphecyTrait;
+use stdClass;
+use Exception;
+
+class GenerateMethodCommandTest extends TestCase
+{
+    use ProphecyTrait;
+
+    public function testSuccessfulCall(): void
+    {
+        $uri = 'file:///file.php';
+        $offset = 5;
+        $textDocumentItem = new TextDocumentItem($uri, 'php', 1, '<?php ');
+        $textDocumentUri = TextDocumentUri::fromString($textDocumentItem->uri);
+        $sourceCode = SourceCode::fromStringAndPath(
+            $textDocumentItem->text,
+            $textDocumentUri->path()
+        );
+        $textEdits = new TextDocumentEdits(
+            $textDocumentUri,
+            new TextEdits(TextEdit::create(5, 1, 'test'))
+        );
+
+        $rpcClient = new class() implements RpcClient {
```

Comment:
By doing a mock `RpcClient` we can intercept calls to `notification` and `request` and then spy on them on mock responses.

Reply:
you can use the test language server builder for this I think

Reply:
https://github.com/phpactor/language-server-phpactor-extensions/blob/adf9f1857732cb0e1365d38f7614313ccf0f8f7a/tests/LanguageServerCodeTransform/Unit/CodeAction/CreateClassProviderTest.php#L65

the transmitter has all the messages that were transmitted from the LS

Reply:
Did it using `TestRpcClient`.